doc/developer/design/20230512_mir_cost_model.md

```diff
@@ -102,6 +104,25 @@ This hasn't been implemented yet.
 # Rollout
 [rollout]: #rollout
 
+## Implementation plan
+[testing-and-observability]: #implementation-plan
+
+The current plan is to work towards a minimally viable cost model, with the following priority ordering memory > hydration > latency > scalability.
+
+- Start with a source cardinality analysis that maps (symbolic) input cardinalities to (symbolic) output cardinalities.
+
+  + Cardinality analysis can work on MIR or LIR, since cardinality ought to be constant under optimization.
+
+  + Selectivity factors will need to be just made up for now.
+
+- Use the cardinality analysis to build the memory cost model.
+
+  + The memory cost model will work on LIR using the "lower and check cost" approach when planning joins in MIR.
+
+- Try to validate the memory cost model on a variety of queries.
```

Comment:
@uce we discussed with @mgree that I might be able to help out with this and the following point (validation / experimentation / fitting) with focus on the `Join` memory estimates after I wrap up / ramp down on the WMR rollout.

I'm not sure how high would this be on the priority list, but at the very least I would like to have it there as an option.